config/clients/go/template/fga_test.mustache

```diff
@@ -1159,7 +1159,7 @@ func Test{{appShortName}}Api(t *testing.T) {
 			RequestPath:    "check",
 		}
 		requestBody := CheckRequest{
-			TupleKey: &TupleKey{
+			TupleKey: TupleKey{
 				User:     PtrString("user:81684243-9356-4421-8fbf-a4f8d36aa31b"),
 				Relation: PtrString("viewer"),
```

Comment:
If we are improving the API proto to more correctly indicate required vs not required fields, why are these still pointers? They are all required in Check.

Or is this because of the old issue that they all share the same `TupleKey` proto interface so that they must be optional on all?

cc @miparnisari @jon-whit

Reply:
Good question ...  Will ask @miparnisari and @jon-whit to investigate.  I believe it is using TupleKey which is shared by read.

Reply:
Because our API overloads the usage of `TupleKey` all over the place and thus the field validations of the TupleKey must be generic without specific constrains based on the API endpoint context they are used for.

For example, the `Read` API takes a TupleKey but only some fields (and sub-strings of the fields for that matter) are required while all of the field are required for the same TupleKey structure in the `Check` API.

This is why I want to change our API from

```
message TupleKey {
    string object = 1;
    string relation = 2;
    string user = 3;
}

message CheckRequest {
    ...
    TupleKey tuple_key = 2;
}

message ReadRequest {
    TupleKey tuple_key = 1;
}
```
all over the place to more specifically and strongly typed and specific

```
message Object {
    string type = 1;
    string id = 2;
}

message RelationshipTuple {
    Object object = 1;
    string relation = 2;
    UserRef user = 3;
}

message CheckRequest {
    RelationshipTuple tuple = 1;
}

message ReadRequest {
    Object object = 1;
    string relation = 2;
}
```

Just for example.

Reply:
Yes, we know that, which is why I asked because I thought solving this was part of the ticket being worked on.

Considering releasing this is already a breaking change, would it make sense to solve the issue with the tuple keys before releasing new versions of the SDKs?

We know about it and it has been on our wishlist for a long while. Can we create a ticket and check in with @aaguiarz when we can slot it into prioritization?

It will be pretty intensive change for the BE so we can optionally release this version with minor breaking changes and then bump it up again when that work is done.